src/ser/mod.rs

```diff
@@ -179,11 +181,11 @@ impl Serializer {
 
     fn end_indent(&mut self) {
         if let Some((ref config, ref mut pretty)) = self.pretty {
+            pretty.indent -= 1;
             if pretty.indent < config.depth_limit {
                 self.output
                     .extend((1..pretty.indent).map(|_| config.indentor.as_str()));
```

Comment:
isn't this changed now? I think you'd want to do `0..pretty.indent` if you subtract 1 earlier

Reply:
Good point.

Testing it out just now, I think the mechanism is insufficient to handle empty sequences and maps because deeper nested items will still have a `pretty.indent > 0`

I moved it around, because the test failed with extra white-space.

Reply:
But this would break the non-empty sequences, right?

Reply:
Yeah, so I've moved it back in the latest iteration, added test data to handle deeper nested empty data sets.

Reply:
I just didn't see the code update, hence the confusion :)